apps/site/assets/ts/helpers/fetch-json.ts

```diff
@@ -0,0 +1,23 @@
+export const isFetchError = <T>(r: FetchResult<T>): r is FetchError =>
+  typeof r === "object" && "statusText" in r;
+
+export const throwIfFetchFailed = <T>(result: FetchResult<T>): T => {
+  if (isFetchError(result)) {
+    throw new Error(`Fetch failed: ${result}`);
+  }
+
+  return result;
+};
+
+export type FetchError = { status: number; statusText: string };
+export type FetchResult<T> = T | FetchError;
+export const fetchJson = async <T>(
+  ...args: Parameters<typeof fetch>
+): Promise<FetchResult<T>> => {
+  const resp = await fetch(...args);
+  if (resp.status !== 200) {
+    return resp;
+  }
+
```

Comment:
typically I prefer not throwing in "library" functions by default since thrown exceptions are not typechecked, and this makes it explicit that the caller needs to handle the error. Definitely see the point here that it's kind of annoying having to call if the caller wants to throw though :) 

how do you feel about adding another export to this file, something like `fetchJsonOrThrow`, which has this behavior by default?

Reply:
> how do you feel about adding another export to this file, something like fetchJsonOrThrow, which has this behavior by default?

I like this idea!